frontend/src/store/index.js

```diff
@@ -532,48 +532,50 @@ const getters = {
   projectNamesFromProjectList (state, getters) {
     return map(getters.projectList, 'metadata.name')
   },
-  customFieldsListShoot (state, getters) {
-    return map(getters.customFieldsShoot, (customFields, key) => {
+  shootCustomFieldList (state, getters) {
+    return map(getters.shootCustomFields, (customFields, key) => {
       return {
         ...customFields,
         key
       }
     })
   },
-  customFieldsShoot (state, getters) {
-    let customFieldsShoot = get(getters.projectFromProjectList, 'metadata.annotations["dashboard.gardener.cloud/customFieldsShoot"]')
-    if (!customFieldsShoot) {
-      return undefined
+  shootCustomFields (state, getters) {
+    function someObjectValuesAreObject () {
+      return object => some(object, isObject)
+    }
+
+    let shootCustomFields = get(getters.projectFromProjectList, 'metadata.annotations["dashboard.gardener.cloud/shootCustomFields"]')
+    if (!shootCustomFields) {
+      return
     }
 
     try {
-      customFieldsShoot = JSON.parse(customFieldsShoot)
+      shootCustomFields = JSON.parse(shootCustomFields)
     } catch (error) {
       console.error('could not parse custom fields', error.message)
-      return undefined
+      return
     }
 
-    customFieldsShoot = pickBy(customFieldsShoot, customFields => {
-      return !isEmpty(get(customFields, 'path')) &&
-        !isEmpty(get(customFields, 'name'))
-    })
+    shootCustomFields = omitBy(shootCustomFields, isEmpty) // omit null values
+    shootCustomFields = omitBy(shootCustomFields, someObjectValuesAreObject())
+    shootCustomFields = pickBy(shootCustomFields, ({ path, name }) => name && path)
```

Comment:
```suggestion
    shootCustomFields = filter(shootCustomFields, customField => {
      if (isEmpty(customField) {
        return false // omit null values
      }
      if (some(customField, isObject)) {
        return false // omit custom fields with object values
      }
      return customField.name && customField.path
    }) 
```

Reply:
filter returns an array, but the object should be preserved